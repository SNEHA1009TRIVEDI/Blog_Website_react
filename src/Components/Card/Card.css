.card {
    box-sizing: border-box;
    outline: 1px solid gray;
    position: relative;
    width: 320px;
    height: 500px;
    display: flex;
    flex-direction: column;
  }
  
  .card > * {
    flex: none;
  }
  
  .card > .vertical-spacer {
    flex: 1;
  }
  
  /* .card-img {
    width: 320px;
    height: 100px;
    object-fit: cover;
  } */

  .card-img {
    width: 100%;
    height: 50%;
    object-fit: cover;
  }
  
  .card-meta-info {
    color: gray;
    font-size: 'small';
    display: flex;
    justify-content: space-between;
    margin: 15px;
  }
  
  .card-padding {
    padding: 2px 12px;
  }
  
  .card-margin {
    margin: 8px 12px;
  }
  
  .card-title {
    font-size: 1.5rem;
    font-weight: bold;
    margin: 0;
  }
  
  .card-content {
    max-height: 57px;
    margin: 5px;
    overflow: hidden;
  }
  
  .card-actions {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  
  .card-actions-icon {
    width: 24px;
    height: 24px;
  }
  