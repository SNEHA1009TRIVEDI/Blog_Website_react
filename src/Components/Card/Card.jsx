import React, { useEffect,useContext } from "react";
import Clap from "../Clap/Clap";
import "./Card.css";
import heart_red from "../../assets/Icons/heart-red.svg";
import heart_black from "../../assets/Icons/heart-black.svg";
import clap from "../../assets/Icons/clapping.svg";
import makeRequest from "../../Utils/makeRequest";
import { BlogPostContext } from "../../contexts/BlogPostContext";
const Card = ({
  id,
  image,
  date,
  reading_time: readingTime,
  title,
  claps,
  liked,
  description,
  imgAlt = "",
}) => {
  const [isLiked, setIsLiked] = React.useState(liked);
  const [count, setCount] = React.useState(claps);
  // const [isLiked, setIsLiked] = React.useContext(liked);
  // const [count, setCount] = React.useContext(claps);
  const { allBlogData, setAllBlogData } = useContext(BlogPostContext);

  // useEffect(() => {
  //    axios({
  //     method: "put",
  //     url: "http://localhost:8080/blog-posts/1",
  //     data: {
  //       claps: count,
  //       liked: isLiked,
  //     },
  //   });
  // }, [count,isLiked]);

  // useEffect(() => {
  //   makeRequest("PUT", "http://localhost:8080/blog-posts/1", count, isLiked);
  // }, []);

  const handleLike = () => {
    makeRequest(
      "PUT",
      `http://localhost:8080/blog-posts/${id}`,
      count,
      !isLiked
    );
    setIsLiked(!isLiked);
  };

  const heartSrc = isLiked ? heart_red : heart_black;

  const handleClick = () => {
    // console.log("clap",count);
    makeRequest(
      "PUT",
      `http://localhost:8080/blog-posts/${id}`,
      count + 1,
      isLiked
    );
    setCount(count + 1);
  };

  return (
    <div className="card" data-testid="blog-post">
      {console.log("../../assests/Images" + image)}
      <img src={image} alt={imgAlt} className="card-img" />
      <p className="card-meta-info card-padding">
        <span>{readingTime}</span>
        <span>{date}</span>
      </p>
      <h3 className="card-title card-padding">{title}</h3>
      <p className="card-content card-padding">{description}</p>

      <div className="vertical-spacer"></div>

      <hr className="card-margin" />
      <div className="card-padding card-actions">
        <div className="card-actions" id="clap">
          <Clap iconPath={clap} clapCount={handleClick} />
          <span>{count}</span>
        </div>

        <div className="card-actions" id="heart">
          <Clap
            iconPath={heartSrc}
            // isLiked={handleLike}
            clapCount={handleLike}
          />
        </div>
      </div>
    </div>
  );
};

export default Card;
