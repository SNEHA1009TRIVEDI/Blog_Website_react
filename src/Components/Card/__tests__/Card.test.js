import { render } from "@testing-library/react";
import React from "react";
import Card from "../Card.jsx";

describe("Card", () => {
  it("should renders without crashing ", () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });
});
