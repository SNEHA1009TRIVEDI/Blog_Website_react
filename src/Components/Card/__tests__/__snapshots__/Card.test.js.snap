// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should renders without crashing  1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="blog-post"
  >
    <img
      alt=""
      class="card-img"
    />
    <p
      class="card-meta-info card-padding"
    >
      <span />
      <span />
    </p>
    <h3
      class="card-title card-padding"
    />
    <p
      class="card-content card-padding"
    />
    <div
      class="vertical-spacer"
    />
    <hr
      class="card-margin"
    />
    <div
      class="card-padding card-actions"
    >
      <div
        class="card-actions"
        id="clap"
      >
        <img
          alt="clap"
          class="card-actions-icon"
          height="25"
          src="clapping.svg"
          width="25"
        />
        <span />
      </div>
      <div
        class="card-actions"
        id="heart"
      >
        <img
          alt="clap"
          class="card-actions-icon"
          height="25"
          src="heart-black.svg"
          width="25"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
